src/dev/formatWhitespace.m

```diff
@@ -0,0 +1,66 @@
+function formatWhitespace()
+    % FORMATWHITESPACE Formats whitespace in the active document
+    %
+    %   Rules:
+    %       - Smart-indent
+    %       - Indent content of all function with 4 spaces per level
+    %       - Remove whitespace in empty lines
+    %       - Preserve indentantion after line continuations, i.e. ...
+
+    import matlab.desktop.editor.*
+    try
+        d        = getActive();
+        oldLines = textToLines(d.Text);
+    catch
+        error('formatWhitespace:noEditorApi','Check that the Editor API is available.')
+    end
+
+    % Parse file into a tree
+    tree = mtree(d.Text);
```

Comment:
I knew this kind of trickery existed somewhere in MATLAB. :smile: Is there actually some sort of documentation that you use for this? If so, I would really love to see it.